### DISCUSSION TITLE
Quick guide to steam sdk

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#2122] [General]
Hello,
as posted in discord, here's a quick guide on how to use Steam's controler api, or least, some of the key points that could save you some time.

1. install the sdk. For this you have to register in steamworks https://partner.steamgames.com/. From there you download the last sdk. Then depending on what idle and OS you are using, you have to set the paths to include the libraries.
2. copy the .dll in the output directory(where the executable is after the build). For instance copy steam_api64.dll in the Debug or Release folder.
3. Again, in the output directory create a text file called steam_appid.txt .Steam provides a test id which is 480.
so the content of  write 480 and save the file.
(To create you own game, you should be a steam partner, and pay 100$ for each game you do. More info on their web)

At this point steam should be integrated with axmol. Now let's move on to how to set up a controller.
1. create the folder controller_config in steam installation path (C:\Program Files (x86)\Steam\controller_config), and inside of that folder the file game_action_X.vdf where X is the app id. E.g game_action_480.
2. copy paste in that .vdf any blue print https://partner.steamgames.com/doc/features/steam_controller/iga_examples
3. have steam running
4. in the AppDelegate::applicationDidFinishLaunching() add the following checks

```///////////////////////////////
    // initialize steam api here //
    ///////////////////////////////

    // 1. initialize Steam API 
    if (!SteamAPI_Init())
    {
        AXLOG(">>> Failed to initialize SteamAPI. Ensure Steam app is launched and that 'steam_appid.txt' is present in \n"
        "the output directory. For instance axmolProject/build/bin/axmolProject/Debug (or /Release)");
        return false;
    }
    else {  AXLOG(">>> SteamAPI initialized successfully");  }

    // 2. Initialize Steam Input
    if(SteamInput())
    {
        SteamInput()->Init(true);
        AXLOG(">>> SteamInput initialized successfully");
    }
    else {  AXLOG(">>> Failed to initialize SteamInput");    }
    //...
```

If no errors are logged at this point, its going well 
from there you can star explore the API
For instance:

```
void HelloWorld::handleSteamInput() {

    if (!SteamInput())
    {
        log(">> failed");
        return;
    }

    InputActionSetHandle_t inGameControlsActionSetHandle = SteamInput()->GetActionSetHandle("InGameControlsActionSet");
    log(">>>>  %llu", inGameControlsActionSetHandle);
}

void HelloWorld::update(float delta)
{
    handleSteamInput();
}
```

Hopeful this will help any one trying to use steam with axmol. 
cheers

**Comment by aismann** [comment]
Is the Wiki not the better long term place?

**Reply by danialias** [reply]
Sadly we can't include guides on how to install extensions or SDKs for external services in the wiki, as they are out our control and may change at any time, and there are hundreds of them. Having this information searchable here in Discussions is the best solution for now.

**Comment by rudiHammad** [comment]
Topic moved to https://github.com/rudiHammad/AxmolSteamInput

**Reply by danialias** [reply]
Thank to this now we are linking this kind of guides as "External tutorials" inside the Tutorials page of the wiki. Thank you for the great guide!
